tests/validation/tests/v3_api/test_bkp_restore.py

```diff
@@ -29,16 +42,25 @@ def test_bkp_restore():
     p_client.create_ingress(name=name,
                             namespaceId=ns.id,
                             rules=[rule])
-    validate_ingress(namespace["p_client"], namespace["cluster"],
-                     [workload], host, path)
+    validate_ingress(p_client, cluster, [workload], host, path)
 
     # Perform Backup
     backup = cluster.backupEtcd()
     backupname = backup['metadata']['name']
+    wait_for_backup_to_active(cluster, backupname)
+
+    # Get all the backup info
     etcdbackups = cluster.etcdBackups(name=backupname)
     etcdbackupdata = etcdbackups['data']
     backupId = etcdbackupdata[0]['id']
-    wait_for_backup_to_active(client, cluster, backupname)
+
+    backupfilename = ""
+    if backup_mode == "s3":
+        backupfileurl = etcdbackupdata[0]['filename']
+        # Check the backup filename exists in S3
+        parseurl = urlparse(backupfileurl)
+        backupfilename = os.path.basename(parseurl.path)
+        assert AmazonWebServices().s3_backup_check(backupfilename)
```

Comment:
Can we do a similar check for local snapshot as well ? In this case we should check for the snapshot to be present as s file in all  etcd nodes?

Reply:
That would require ssh into each of the etcd nodes' filesystem and look for the filename, that's beyond the scope of this PR. I can work on that in a separate PR as it doesn't sound a trivial addition.